bigtable/admin.go

```diff
@@ -900,6 +895,9 @@ func (iac *InstanceAdminClient) Instances(ctx context.Context) ([]*InstanceInfo,
 			Labels:        i.Labels,
 		})
 	}
+	if len(res.FailedLocations) > 0 {
+		return is, status.Errorf(codes.Unavailable, "Failed locations: %v", res.FailedLocations)
```

Comment:
Add a comment to be explicit that we are returning partial results in spite of some locations being unavailable.

Reply:
I think for something like this to work some sort of sentinel error would need to be returned. I think most people would just see an err is non-nil and not continue further processing. If we do something like this we would want to document it clearly. An example of something like this in the std lib would be [Reader](https://golang.org/pkg/io/#Reader)

Reply:
Sounds reasonable. Pushing more comments and new error type